Improve text of README.md (#905)
Fantom-foundation/Carmen: README.md

# Introduction

> Carmen is a fast and space conservative database for blockchains. It outperforms other projects in the

Reviewer: I think this two original lines should be (in some form) kept. It is our pitch - the speed that outperforms others and the space is a fraction of what provides others. 

The new pitch is way too complicated, it suddenly speaks about balances, code, smart contract.

Author: @kjezek  - can you rewrite it and submit it? I can check after.

Reviewer: done, please have a look

Author: We should focus on functionality rather than non-functional properties.

> Carmen is a high-performance data store for Blockchains and manages balances, code, and the persistent storage of smart contracts. Carmen assumes a linear evolution of blocks. Carmen supports two versions of databases: LiveDB, which keeps the last state of the last block only, and ArchiveDB, which keeps all states over all blocks. The Carmen project implements various schemas (currently S1 to S5), including a Merkle-Patricia Tries (MPT) variation as an underlying data structure for storing state.  The storage layer is abstracted, and the schemas can read and write information in memory, a key-value store (LevelDB, etc.), or a native file format. Carmen has implemented schema S1-S3 in the Go and C++ languages. Schema S4 and S5 are implemented only in Go.  The latest Schema S5 is currently used in the Sonic Client.

Reviewer: ```suggestion
Carmen is a high-performance data store for Blockchains and manages balances, code, and the persistent storage of smart contracts. Carmen assumes a linear evolution of blocks. Carmen supports two versions of databases: LiveDB, which keeps the last state of the last block only, and ArchiveDB, which keeps all states over all blocks. The Carmen project implements various schemas (currently S1 to S5), including a Merkle-Patricia Trie (MPT) variation as an underlying data structure for storing state.  The storage layer is abstracted, and the schemas can read and write information in memory, a key-value store (LevelDB, etc.), or a native file format. Carmen has implemented schema S1-S3 in the Go and C++ languages. Schema S4 and S5 are implemented only in Go.  The latest Schema S5 is currently used in the Sonic Client.
```

Reviewer: Could you also add line separators so it is not a one long line?

# How to Use

> Schema S5 is the only schema compatible with Merkle-Patricia Tries. Carmen has always enabled LiveDB to keep the state of the last block.

Reviewer: ```suggestion
Schema S5 is the only schema compatible with Merkle-Patricia Tries. Carmen enables always LiveDB to keep the state of the last block.
```

ArchiveDB can be enabled to retain historical data. If ArchiveDB is enabled, the state of the last block is appended to the archive for later retrieval. 

> The Carmen interface is loosely related to the go-ethereum's StateDB interface, though it differs w.r.t. block management and transaction management. 

Reviewer: not sure if this change improves this paragraph. First it IMHO unnecessarily refers to ethereum's StateDB and then suddenly it becomes too brief

## Query API

> LiveDB and Archive can be queried using a **query API** using a functional style callback as shown below:

Reviewer: ```suggestion
LiveDB and Archive can be queried using a **query API** providing a functional style callback as shown below:
```

Reviewer: avoided `using` twice in one sentence

# Performance 

Carmen is the StateDB of [Fantom Sonic](https://fantom.foundation/sonicPage) blockchain. 

> The data store is the key component for enabling low time-to-finally and high transaction throughput while saving disk space. 

Reviewer 2: ```suggestion
The data store is the key component for enabling low time-to-finality and high transaction throughput while saving disk space. 
```
